components/extensions/org.wso2.carbon.ml.siddhi.extension/src/main/java/org/wso2/carbon/ml/siddhi/extension/PredictStreamProcessor.java

```diff
@@ -203,6 +274,25 @@ private void populateFeatureAttributeMapping() throws Exception {
         }
     }
 
+    private Object getMostFrequent(Object[] predictions) {
```

Comment:
commons-math mode only works for numerical values.
Used [apache.commons.lang3.ObjectUtils.mode](https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/ObjectUtils.html). This does essentially the same thing done in the method.